### DISCUSSION TITLE
improvement of the model specification

### DISCUSSION THREAD

**Discussion by SvenjaSchwarck** [cdriveraus/ctsem#21] [Q&A]
Hi Charles,
we analyse time series of physiological and cognitive data, but also hypothesize a systematic relationship. We started with this model specification, which is primarily based on the defaults; We have only two latent variables with two manifest variables each. Both cross-effects of A are set to indvarying. We are primarily interested in the cross-effect drift_COG_PP. 

 our.model<-ctModel(type='stanct',
                       n.latent=2, latentNames=c('PP','COG'),
                       n.manifest=4,
                       manifestNames=c("mPP_1", "mPP_2",
                                       "mCOG_1", "mCOG_2"),
                  LAMBDA=matrix(c(1,1,0,0,0,0,1,1), nrow = 4, ncol=2))

Our question is how to get the stochastics into our model? Our current model saturates very quickly; the current cross-effects go down to zero after a few days. Though our observations are dynamically more complex than our current model.  We were wondering if, for example, our noise is too small and we need to change the default specification at model$pars$transform? Do you have any suggestions?

And a more general question: is the A matrix alone time dependent? According to the subject level latent model equation only Eta is, right?

Best,
Svenja

**Comment by cdriveraus** [comment]
Stochastics are in by default, if I understand your question correctly. You have free parameters in the latent diffusion matrix and the observation error matrix. 

![image](https://user-images.githubusercontent.com/4108549/140509618-58c30f05-eedd-4f56-bd5f-2447a9d66b02.png)

I don't really know what "Though our observations are dynamically more complex than our current model" means, sorry, can you elaborate on the issue? If you have lots of data and a more complex system (first order systems, ie the default, are pretty simple really!) you may need to build in higher order processes and or faster / slower components. 

The A matrix is fixed over time by default. You can induce dependencies in this, see the non-linear section of the vignette for a basic guide / example.

**Reply by SvenjaSchwarck** [reply]
The question goes along the following ideas. We have set up a first order model with two state variables (fully connected) describing session to session variability over 70 observations (within 6 months) in 19 subjects.
The state equation with parameters and diffusion term is estimated and the fitting seems to work.
However, the time series with quite a bit of (potentially systematic or random) fluctuations over half a year seems not to be fitted very well.
There seem to be some early dynamical changes of the state variables only at the first few sessions (following the usual exponential dynamics), but then (very fast) the system directly just decays to 'zero' state stability and stays there for the remaining >65 timepoints.
Our naive explanation at the moment is that either the stochastic part, which we would expect to push the states randomly out of
the 'zero' state is not fitted properly (ie. our system is more or less deterministic) or these first order models do not describe the specific dynamics of our physical performance and cognitive performance states and their interactions over these 70 sessions very well.

Do you have any ideas what could be the reason for this 'early stability' and the system not capturing the session to session fluctuations or potential recommendations to improve on this ? Maybe we also plot the wrong thing, but it seems that eta(t) decays to 0 rather than fluctuating over the whole timeseries.

**Reply by cdriveraus** [reply]
Maybe you want to post whatever output you're using that leads to these conclusions? From the sound of what you describe I envisage a process that is mostly just unpredictable / measurement noise, but that would be unusual even for social science fields! :)

**Reply by SvenjaSchwarck** [reply]
First, as an overview:


![model_latex_2CR](https://user-images.githubusercontent.com/93767420/142008259-9d2b3c49-7ef7-4edc-ac45-db8475335be1.png)


Regarding the output, we were interested in the DRIFT results of the  summary(model.fit)$popmeans output:

drift_PP	-0.3299	0.0353	-0.4035	-0.3287	-0.2663
drift_PP_COG	-0.0745	0.0565	-0.1813	-0.0756	0.0411
drift_COG_PP	-0.1897	0.0975	-0.3872	-0.1875	-0.0016
drift_COG	-0.9167	0.1975	-1.3424	-0.9033	-0.5838

In a second step, we plotted the data using:
ctStanDiscretePars(model.fit, plot =T, times = seq(0,70), indices = 'CR')
resulting in following discrete-time cross-effect dynamics plot:

![2CR-discrete_time_plot](https://user-images.githubusercontent.com/93767420/142007812-8651f6a2-c87b-4bdc-85c1-aa2da93e5dba.png)



If we interpret this plot correctly, the effect of PP on COG peaks at a time interval of ~ 1-4 days and then decays relatively fast to ‘zero’  and stays there for the remaining > 40 days.

**Reply by cdriveraus** [reply]
Ok, this plot doesn't say anything about the stochastic elements, it just shows the expectations given some impulse at time zero. According to the fitted model, a random change in pp can improve predictions of cog for up to around 20 days, after which the stochastic elements dominate. This looks pretty normal for a first order model to me, do you have strong reason to expect differently? Also, you can set the argument observational=T when plotting to visualise the expected pattern after a typical observed change (considering correlation in random changes), rather than a hypothetical experimental one....

**Reply by cdriveraus** [reply]
The ctKalman plots probably show you better how the model is fitting by the way -- these show the actual model predictions conditioned on your data, not just conditioned on a single random change.

**Reply by SvenjaSchwarck** [reply]
Thank you for the clarification. Good to hear that this looks pretty normal for our first order model :)
And thank you for the advice regarding the ctKalman plot  - this one really shows better how our model is fitting.
